google/cloud/forseti/scanner/scanners/retention_scanner.py

```diff
@@ -116,9 +119,18 @@ def _retrieve_bucket(self):
         all_lifecycle_info = []
         with scoped_session as session:
             for bucketinfo in data_access.scanner_iter(session, 'bucket'):
-                lifecycle_info = retention_bucket.RetentionBucket.from_json(
-                    bucketinfo)
-                all_lifecycle_info.append(lifecycle_info)
+                bucketdatadict = json.loads(bucketinfo.data)
+                lifecycleitems = []
+                if ('lifecycle' in bucketdatadict and
+                        'rule' in bucketdatadict.get('lifecycle')):
+                    lifecycleitems = bucketdatadict['lifecycle']['rule']
+
+                new_bucket_res = resource_util.create_resource(
```

Comment:
You can just create the Bucket resource directly without using resource_utils, then you can pass the lifecycle in the constructor.

Alternatively, add a from_json method (similar to other Resources), then you can just use something like:

`return bucket.Bucket.from_json(parent=parent_proj, bucket_info.data)`